src/video/SDL_shape.c

```diff
@@ -75,24 +75,10 @@ void SDL_CalculateShapeBitmap(SDL_WindowShapeMode mode, SDL_Surface *shape, Uint
     for (y = 0; y < shape->h; y++) {
         bitmap_scanline = bitmap + y * bytes_per_scanline;
         for (x = 0; x < shape->w; x++) {
-            alpha = 0;
-            pixel_value = 0;
-            pixel = (Uint8 *)(shape->pixels) + (y * shape->pitch) + (x * shape->format->BytesPerPixel);
-            switch (shape->format->BytesPerPixel) {
-            case (1):
-                pixel_value = *pixel;
-                break;
-            case (2):
-                pixel_value = *(Uint16 *)pixel;
-                break;
-            case (3):
-                pixel_value = *(Uint32 *)pixel & (~shape->format->Amask);
-                break;
```

Comment:
I am suspicious about this code, which is why I'm replacing it. If the pointer `pixel` is misaligned, that's undefined behaviour, and on some CPU architectures (mostly RISC), it will either crash or give silently wrong answers (on ARM I think it traditionally gave the result that would have been correct for dereferencing the last correctly-aligned pointer less than or equal to this one).